packages/abi-coder/src/interface.ts

```diff
@@ -118,9 +118,11 @@ export default class Interface {
       return selector;
     }
 
-    const isStruct = !(inputs.length === 1 && inputs[0].type === 'u64');
+    const isReferenceType = !(inputs.length === 1 && inputs[0].type === 'u64');
```

Comment:
Should this not also check for `u8`, `u16`, `u32`, and `bool`, or are those all folded into `u64` earlier?

Reply:
This is one of the things that will be covered in https://github.com/FuelLabs/fuels-ts/issues/268. I will add a TODO here.